f2/src/forms/AnonymousInfoForm.js

```diff
@@ -1,76 +1,65 @@
 import PropTypes from 'prop-types'
-import { useLingui } from '@lingui/react'
 import { Trans } from '@lingui/macro'
 import { Form } from 'react-final-form'
 import { NextAndCancelButtons } from '../components/next-and-cancel-buttons'
-import { RadioAdapter } from '../components/radio'
+import { CheckboxAdapter } from '../components/checkbox'
 import { Stack } from '@chakra-ui/core'
 import { useStateValue } from '../utils/state'
-import { areFieldsValid } from '../utils/areFieldsValid'
 import { FormArrayControl } from '../components/FormArrayControl'
-import { formDefaults } from './defaultValues'
-
-const validate = (values) => {
-  const errors = {}
-  //condition for an error to occur: append a lingui id to the list of error
-  if (!values.anonymous || values.anonymous.length < 1) {
-    errors.anonymous = 'AnonymousPage.warning'
-  }
-  return errors
-}
+import { Alert } from '../components/Messages'
 
 const clearData = (dataOrig) => {
   let data = JSON.parse(JSON.stringify(dataOrig))
   return data
 }
 
 export const AnonymousInfoForm = (props) => {
-  const localOnSubmit = (data) => {
-    if (areFieldsValid(data, formDefaults.anonymous)) props.onSubmit(data)
-  }
-
-  const { i18n } = useLingui()
-
   const [data] = useStateValue()
+  const whetherCheck = {
+    checkBoxOptions: '',
+    ...data.formData.whetherCheck,
+  }
+  const checkBoxOptions = ['anonymousPage.yes']
   const anonymous = {
-    ...formDefaults.anonymous,
+    checkBoxOptions: '',
     ...data.formData.anonymous,
   }
 
-  const ifanonymous = ['anonymousPage.yes', 'anonymousPage.no']
-
   return (
     <React.Fragment>
       {false ? ( // mark ids for lingui
         <div>
           <Trans id="anonymousPage.yes" />
-          <Trans id="anonymousPage.no" />
         </div>
       ) : null}
 
       <Form
         initialValues={anonymous}
-        onSubmit={(data) => localOnSubmit(clearData(data))}
-        validate={validate}
+        onSubmit={(data) => props.onSubmit(clearData(data))}
         render={({ handleSubmit }) => (
           <Stack
             as="form"
             onSubmit={handleSubmit}
             shouldWrapChildren
             spacing={12}
           >
-            <FormArrayControl
-              name="anonymous"
-              label={<Trans id="anonymousForm.title" />}
-              errorMessage={<Trans id="anonymousPage.warning" />}
-            >
-              {ifanonymous.map((key) => {
+            <FormArrayControl name="checkBoxOptions">
+              {checkBoxOptions.map((key) => {
                 return (
                   <React.Fragment key={key}>
-                    <RadioAdapter name="anonymous" value={key}>
-                      {i18n._(key)}
-                    </RadioAdapter>
+                    <CheckboxAdapter
+                      name="checkBoxOptions"
+                      value={key}
+                      isChecked={whetherCheck.checkBoxOptions.includes(key)}
+                      conditionalField={
```

Comment:
This works, but is not a conditional field. The alert should be moved below the field, with its own conditional